### PR TITLE
Use bazel-distribution for `assemble-maven`

### PR DESCRIPTION
**Pull request by flyingsilverfin**
Following the implemation of https://github.com/graknlabs/bazel-distribution/pull/248, we can remove templated `assemble-maven` from `dependencies`. This removes the `distribution/maven` targets and notably removes a call to templating using `deployment_rules_builder`. `//library/maven:artifactz.bzl:maven_overrides` is added as a method to take the repository maven overrides and merge them with repo maven overrides.

### CODE REVIEW

File: dependencies/graknlabs/repositories.bzl [hunk 1]
```diff
@@ -16,15 +15,11 @@
...
-bazel build //distribution/maven:rules
-install -m 644 $(bazel info bazel-genfiles)/distribution/maven/rules.bzl $WORKSPACE_HOME/distribution/maven/rules.bzl
-
-popd > /dev/null
+def graknlabs_bazel_distribution():
+    git_repository(
+        name = "graknlabs_bazel_distribution",
+        remote = "https://github.com/graknlabs/bazel-distribution",
+        commit = "30e30cec9e3fe4821103cafbd240ee9862e262ea" # sync-marker: do not remove this comment, this is used for sync-dependencies by @graknlabs_bazel_distribution
+    )
```
Comment (vmax) [inline dependencies/graknlabs/repositories.bzl:25]: @flyingsilverfin newline please

File: library/maven/artifacts.bzl [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+    overriden = {}
+    for artifact in normalised_artifacts_org:
+        overriden[artifact] = maven_overrides_repo.get(artifact, normalised_artifacts_org[artifact])
+    return overriden
+
+def _artifact_versions(artifacts_with_excludes):
+    """ Normalise the provided master maven artifacts versions to just artifact : version """
+    just_versions = {}
+    for artifact, value in artifacts_with_excludes.items():
+        if type(value) == type({}):
```
Comment (vmax) [inline library/maven/artifacts.bzl:21]: @flyingsilverfin would this also work?
```suggestion
        if type(value) == dict:
```

Comment (flyingsilverfin) [reply library/maven/artifacts.bzl:21]: yep will do

File: dependencies/graknlabs/repositories.bzl (hunk 1 above)
Comment (flyingsilverfin) [reply dependencies/graknlabs/repositories.bzl]: done

**Pull request merged**